### PR TITLE
metrics: cleans k8s iperf deployment when the test finishes.

### PR DESCRIPTION
**Pull request by dborquez**
This PR fixes small issues like:
1. Cleaning up the k8s environment by removing the iperf test implementation even when the test fails.
2. Checks if the workload returned a result before generating an empty results json file as it was bein done.
3. Removes the redundancy of calls to functions that process subtests and should compose the results json file only when all results are ready and not before.
4. The tcp service manifest was added to the server deployment which targets TCP port 5201.

Fixes: #8534

### CODE REVIEW

**Review by GabyCT — APPROVED** [review]

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh [hunk 1]
```diff
@@ -100,79 +99,77 @@ function iperf3_jitter() {
...
-		metrics_json_init
-		metrics_json_start_array
-
-		local json="$(cat << EOF
-		{
-			"jitter": {
-				"Result" : $jitter_result,
-				"Units" : "ms"
-			}
+	[ -z "${jitter_result}" ] && die "Iperf3 was unable to collecte Jitter results."
```
Comment (chavafg) [inline tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:102]: s/collecte/collect/

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh [hunk 2]
```diff
@@ -29,37 +29,36 @@ COLLECT_ALL="${COLLECT_ALL:-false}"
...
 IPERF_DAEMONSET="${SCRIPT_PATH}/runtimeclass_workloads/iperf3-daemonset.yaml"
 
-function remove_tmp_file() {
-	rm -rf "${iperf_file}"
-}
+function iperf3_all_collect_results() {
 
-trap remove_tmp_file EXIT
+	if [ -z "${bandwidth_result}" ] || [ -z "${jitter_result}" ] || [ -z "${cpu_result}" ] || [ -z "${parallel_result}" ]; then
+		die "iperf has not results to save"
```
Comment (chavafg) [inline tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:35]: suggested rewording: iperf couldn't find any results to save

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh [hunk 3]
```diff
@@ -100,79 +99,77 @@ function iperf3_jitter() {
...
+	[ "$COLLECT_ALL" == "true" ] && return
+
+	metrics_json_init
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"jitter": {
+			"Result" : "${jitter_result}",
+			"Units" : "ms"
```
Comment (chavafg) [inline tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:112]: s/`"ms"`/`"${jitter_units}"`

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh [hunk 4]
```diff
@@ -100,79 +99,77 @@ function iperf3_jitter() {
...
 )"
-		metrics_json_add_array_element "$json"
-		metrics_json_end_array "Results"
-	fi
+	metrics_json_add_array_element "${json}"
+	metrics_json_end_array "Results"
 }
 
 function iperf3_cpu() {
 	# Start server
```
Comment (chavafg) [inline tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:148]: is this comment still relevant in this part of the function? iiuc, the iperf server is started in the `iperf3_start_deployment` function. I see this same comment in other functions.

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh (hunk 3 above)
Comment (dborquez) [reply tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:112]: good catch :)

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh (hunk 4 above)
Comment (dborquez) [reply tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:148]: we can't get rid of this comment, thanks @chavafg

File: tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh (hunk 1 above)
Comment (dborquez) [reply tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh:102]: goot catch, thanks @chavafg

File: tests/metrics/network/iperf3_kubernetes/runtimeclass_workloads/iperf3-deployment.yaml [hunk 5]
```diff
@@ -42,3 +42,17 @@ spec:
           name: server
       terminationGracePeriodSeconds: 0
       runtimeClassName: kata
```
Comment (amshinde) [inline tests/metrics/network/iperf3_kubernetes/runtimeclass_workloads/iperf3-deployment.yaml:44]: My suggestion would be be out changes to the deployment yaml in its own separate commit with an explanation of why you are doing so.

Comment (dborquez) [reply tests/metrics/network/iperf3_kubernetes/runtimeclass_workloads/iperf3-deployment.yaml:44]: @amshinde thanks, new commit added.

**Review by amshinde — APPROVED** [review]

**Pull request merged**